pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -721,25 +722,16 @@ def _validate_precision_choice(self) -> None:
 
     def _lazy_init_strategy(self) -> None:
         """Lazily set missing attributes on the previously instantiated strategy."""
-        self.strategy.accelerator = self.accelerator
-        if self.precision_plugin:
-            self.strategy.precision_plugin = self.precision_plugin
-        if self.checkpoint_io:
-            self.strategy.checkpoint_io = self.checkpoint_io
-        if hasattr(self.strategy, "cluster_environment"):
-            self.strategy.cluster_environment = self.cluster_environment
-        if hasattr(self.strategy, "parallel_devices"):
-            if self.strategy.parallel_devices:
-                self._parallel_devices = self.strategy.parallel_devices
-            else:
-                self.strategy.parallel_devices = self._parallel_devices
-        if hasattr(self.strategy, "num_nodes"):
-            self.strategy._num_nodes = self._num_nodes_flag
-        if hasattr(self.strategy, "_layer_sync"):
-            self.strategy._layer_sync = self._layer_sync
-        if hasattr(self.strategy, "set_world_ranks"):
-            self.strategy.set_world_ranks()
-        self.strategy._configure_launcher()
+        self.strategy.lazy_init(
+            accelerator=self.accelerator,
+            precision_plugin=self._precision_plugin,
+            checkpoint_io=self.checkpoint_io,
+            cluster_environment=self.cluster_environment,
+            parallel_devices=self._parallel_devices,
+            num_nodes=self._num_nodes_flag,
+            layer_sync=self._layer_sync,
+        )
+        self._parallel_devices = self.strategy.parallel_devices
```

Comment:
we need actually implement the lazy_init() then change accelerator_connector. 
I think we can have one PR introduce lazy_init in strategies (it will be great to have them in one PR, easier to compare), the second one enable the call to lazy_init()